plugins/modules/vm_portgroup_info.py

```diff
@@ -0,0 +1,162 @@
+description:
+    - Returns information about the standard or distributed portgroups of virtual machines.
+author:
+    - Ansible Cloud Team (@ansible-collections)
+requirements:
+    - vSphere Automation SDK
+options:
+    vm_names:
+        description:
+            - VM names for retrieving the information about portgroup
+        required: true
+        type: list
+        elements: str
+extends_documentation_fragment:
+    - vmware.vmware.vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather list of portgroup by VMs
+  vmware.vmware.portgroup_info:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    vm_names:
+      - vm-test1
+      - vm-test2
+'''
+
+RETURN = r'''
+vm_portgroup_info:
+    description:
+        - Dictionary of the requested VMs with the portgroup information
+    returned: On success
+    type: dict
+    sample: {
+        "vm1": [
+        {
+            "name": "Network Name",
+            "nic_mac_address": "00:00:00:00:00:00",
+            "nic_mac_type": "ASSIGNED",
+            "nic_type": "VMXNET3",
+            "port_id": "network-port-id",
+            "type": "STANDARD_PORTGROUP",
+            "vlan_id": "0",
+            "vswitch_name": "vSwitch0"
+        }]
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.vmware.vmware.plugins.module_utils.vmware import PyVmomi
+from ansible_collections.vmware.vmware.plugins.module_utils import vmware_network
+from ansible_collections.vmware.vmware.plugins.module_utils.vmware_rest_client import VmwareRestClient
+
+
+class PortgroupInfo(PyVmomi):
+    def __init__(self, module):
+        super(PortgroupInfo, self).__init__(module)
+        self.vmware_client = VmwareRestClient(module)
+        self.vms = self.params['vm_names']
+
+    def get_dvs_portgroup_detailed(self, pg_id):
+        dvs_pg = self.get_dvs_portgroup(pg_id)
+        pg = {'portgroup_name': dvs_pg.name, 'vswitch_name': dvs_pg.config.distributedVirtualSwitch.name,
+              'type': 'DISTRIBUTED_PORTGROUP', 'port_id': pg_id,
+              'port_binding': vmware_network.get_dvs_port_allocation(dvs_pg.config.type),
+              'port_allocation': vmware_network.get_dvs_auto_expand(dvs_pg.config.autoExpand),
+              'network_policy': vmware_network.get_dvs_network_policy(dvs_pg.config.defaultPortConfig.macManagementPolicy),
+              'mac_learning': vmware_network.get_dvs_mac_learning(dvs_pg.config.defaultPortConfig.macManagementPolicy.macLearningPolicy),
+              'teaming_policy': vmware_network.get_teaming_policy(dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy),
+              'port_policy': vmware_network.get_port_policy(dvs_pg.config.policy),
+              'vlan_info': vmware_network.get_vlan_info(dvs_pg.config.defaultPortConfig.vlan)
+              }
+
+        if dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy and \
```

Comment:
Just to ease the reading of this code you may create some temp vars like:
```
port_config = dvs_pg.config.defaultPortConfig
```
etc